tests/nested/manual/muinstaller-real/task.yaml

```diff
@@ -244,3 +248,60 @@ execute: |
           remote.exec "sudo cat $DM | jq '.pc.\"structure-encryption\".\"ubuntu-data\".method'" | MATCH '"LUKS"'
       done
   fi
+
+
+  # test kernel/gadget refreshes via the fake-store
+
+  # setup fake store again and make it available inside the nested vm
+  setup_fake_store "$STORE_DIR"
+  export NESTED_FAKESTORE_BLOB_DIR="$STORE_DIR"
+  export NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL="$SNAPPY_FORCE_SAS_URL"
+
+  #shellcheck source=tests/lib/store.sh  
+  . "$TESTSLIB"/store.sh  
+
+  # nested vm needs to restart for fake-store
+  nested_shutdown
+  export NESTED_TPM_NO_RESTART=1
+  nested_start_core_vm
+  unset NESTED_TPM_NO_RESTART
+
+  # make new gadget
+  fakestore make-refreshable --dir "$STORE_DIR" --snap-orig-blob pc.snap.orig --snap-blob ./pc-gadget pc
+
+  # refresh gadget
+  boot_id="$(tests.nested boot-id)"
```

Comment:
Suggestion: maybe reuse the function defined in https://github.com/snapcore/snapd/blob/master/tests/nested/manual/fde-on-classic/task.yaml#L70

Reply:
Thanks, indeed, excellent suggestion. I will update the code to use something similar (the approach is slightly different as this uses the fake store using a function is great).